packages/lesswrong/server/editor/make_editable_callbacks.ts

```diff
@@ -277,10 +277,30 @@ export function dataToMarkdown(data, type) {
   }
 }
 
+/**
+ * When we calculate the word count we want to ignore footnotes. There's two syntaxes
+ * for footnotes in markdown:
+ *
+ * 1.  ^**[^](#fnreflexzxp4wr9h)**^
+ *
+ *     The contents of my footnote
+ *
+ * and
+ *
+ * [^1]: The contents of my footnote.
+ *
+ * In both cases, the footnote must start at character 0 on the line. The strategy here
+ * is just to find the first place where this occurs and then to ignore to the end of
+ * the document.
+ */
 export async function dataToWordCount(data, type) {
   try {
-    const markdown = dataToMarkdown(data, type) || ""
-    return markdown.split(" ").length
+    const markdown = dataToMarkdown(data, type) ?? "";
+    const withoutFootnotes = markdown
+      .replace(/^1\. {2}\^\*\*\[\^\]\(#(.|\n)*/m, "")
+      .replace(/^\[\^\d+\]:.*/m, "");
```

Comment:
Why check for `1` in the first regex and `\d` in the second? I feel like we should be consistent between them. Also it seems like even in the second case, the footnote can spill into multiple lines: https://www.markdownguide.org/extended-syntax/#footnotes

And I guess it doesn't even need to use a number, you can use a string (though this case is prob rare and might be worth skipping).

Reply:
Good point! I've changed the second regex to check for "1" instead. I've ignored the case where they can enter strings for now since it's an edge case as you say, and I think it will make it more likely that we hit false positive matches. I've also added a couple of unit tests for this behaviour.

Reply:
Thanks! I think we'll also want `(.|\n)*` in the second regex as well, since that can include newlines too?